src/Nethermind/Nethermind.TxPool/BlobFinder.cs

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.TxPool.Collections;
+
+namespace Nethermind.TxPool;
+
+public class BlobFinder(BlobTxDistinctSortedPool blobPool)
```

Comment:
Do we even need `BlobFinder` as it uses private parts of `BlobPool`, maybe it should be just under there - method in `BlobTxDistinctSortedPool`?

Reply:
There is one problem. We have all blob index logic in `BlobTxDistinctSortedPool`, but there is also `PersistentBlobTxDistinctSortedPool`, which has a logic of saving/loading blob txs on disc. In `BlobFinder` we are using `TryGetValue`, which has override in `PersistentBlobTxDistinctSortedPool`. If we move a logic from `BlobFinder` to `BlobTxDistinctSortedPool`, we will skip override from `PersistentBlobTxDistinctSortedPool`. By using external class `BlobFinder` and blob pool as an argument, we are always using highest override of `TryGetValue`

Reply:
I had some issues with overrides when implementing blob pool, but maybe it was another case. I'm not fully convinced that it will work like that, but I believe you and let's try. I will remove `BlobFinder` at all

Reply:
It is working perfectly fine, I had completely wrong assumptions. Thanks for caching it!